kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue

```diff
@@ -125,29 +101,78 @@
     components: {
       'icon-button': require('kolibri.coreVue.components.iconButton'),
     },
-    props: [
-      'userid', 'username', 'fullname', 'roles', // TODO - validation
-    ],
+    props: {
+      userid: {
+        type: String, // string is type returned from server
+        required: true,
+      },
+      fullname: {
+        type: String,
+        required: true,
+      },
+      username: {
+        type: String,
+        required: true,
+      },
+      userkind: {
+        type: String,
+        required: true,
+      },
+    },
     data() {
       return {
         username_new: this.username,
         password_new: '',
         password_new_confirm: '',
         fullName_new: this.fullname,
-        role_new: this.roles.length ? this.roles[0].kind : 'learner',
+        kind_new: this.userkind,
```

Comment:
You're saying that I should use the locally scoped prop as the data I'm editing? I think there might be issues with that. 

I have it being set here so that the kind is reflective of what the user already is when you bring up the edit modal. I'm also not sure how good an idea it is to have `v-model` manipulate a `prop` rather than `data`